### PR TITLE
test: Try pinning a lower version in go.mod

### PR DESCRIPTION
**Pull request by jonathan-innis**
Fixes #N/A 

**Description**

**How was this change tested?**

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: go.mod [hunk 1]
```diff
@@ -1,6 +1,6 @@
 module sigs.k8s.io/karpenter
 
-go 1.22.4
+go 1.22.2
```
Comment (fmuyassarov) [inline go.mod:3]: Would you mind to share why?

Comment (jonathan-innis) [reply go.mod:3]: Just a test attempt. Some downstream users of this package were running into some painpoints around upgrading to recently released go versions, so we were trying to see if this is possible. I'm not fully convinced that we should do this yet, though.

**Comment by jonathan-innis** [comment]
Closing since this was just to validate that this works with our vulnchecks.

File: go.mod (hunk 1 above)
Comment (fmuyassarov) [reply go.mod:3]: thanks for sharing

**Pull request closed without merging**